### ISSUE TITLE
--fix for prefer-interface breaks the code, generics get erased

### ISSUE THREAD

**Issue by grumd** [cartant/eslint-plugin-etc#22]
Input:

```ts
export type MyType<
  Foo = Record<string, string>,
  Bar = Record<string, string>,
  Baz = Record<string, string>
> = {
  foo: Foo;
  bar: Bar;
  baz: Baz;
};
```

Expected output:

```ts
export interface MyType<
  Foo = Record<string, string>,
  Bar = Record<string, string>,
  Baz = Record<string, string>
> {
  foo: Foo;
  bar: Bar;
  baz: Baz;
}
```

Actual output:

```ts
export interface MyType {
  foo: Foo;
  bar: Bar;
  baz: Baz;
}
```

**Comment by cartant** [comment]
Thanks for the bug report. Should be fixed in 1.1.2.

**Comment by grumd** [comment]
Lmao that was wicked fast :D
